lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -119,3 +116,29 @@ void migrateClassToMixin(ClassDeclaration node, YieldPatch yieldPatch,
   propsAndStateClassNamesConvertedToNewBoilerplate[originalPublicClassName] =
       newMixinName;
 }
+
+class SemverHelper {
+  Map _exportList;
+
+  SemverHelper(Map jsonReport) {
+    _exportList = jsonReport['exports'];
+  }
+
+  /// Returns semver report information for [node] if it is publicly exported.
+  ///
+  /// If [node] is not publicly exported, returns `null`.
+  Map<String, dynamic> getPublicExportLocations(ClassDeclaration node) {
+    final className = node?.name?.name;
+    String classKey;
+
+    if (className == null || _exportList == null) return null;
+
+    _exportList.forEach((key, value) {
+      if (value['type'] == 'class' && value['grammar']['name'] == className) {
+        classKey = key;
+      }
+    });
```

Comment:
Some of these semver reports are a little large, so I think this could be optimized a little to break once it finds a match:

```dart
for (final key in _exportList.keys) {
  final value = _exportList[key];
      
  if (value['type'] == 'class' && value['grammar']['name'] == className) {
    return value;
  }
}
```